x-pack/packages/kbn-elastic-assistant/impl/assistant_context/index.tsx

```diff
@@ -288,6 +298,9 @@ export const AssistantProvider: React.FC<AssistantProviderProps> = ({
       nameSpace,
       registerPromptContext,
       selectedSettingsTab,
+      // can be undefined from localStorage, if not defined, default to true
+      assistantStreamingEnabled: localStorageStreaming ?? true,
```

Comment:
Doesn't look like you need the fallback here -- the default to `true` above on line #213 should handle the initial read when the key doesn't exist. I just deleted the localstorage key and this fallback and everything worked on a refresh (key created and defaulted to `true`). Or is there a different corner case this is handling?

```suggestion
      assistantStreamingEnabled: localStorageStreaming,
```

Reply:
Because of the `useLocalStorage` type, `localStorageStreaming` is either the type you pass (`boolean`) or `undefined`. Unfortunately, without this fallback you will get the type error down on L361:

```
is not assignable to type UseAssistantContext
Types of property assistantStreamingEnabled are incompatible.
Type boolean | undefined is not assignable to type boolean
Type undefined is not assignable to type boolean
```